signalfx-java/src/main/java/com/signalfx/connection/RetryStrategy.java

```diff
@@ -0,0 +1,25 @@
+package com.signalfx.connection;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.ServiceUnavailableRetryStrategy;
+import org.apache.http.protocol.HttpContext;
+
+public class RetryStrategy implements ServiceUnavailableRetryStrategy {
```

Comment:
Please include a unit test class with complete branch coverage for this new class.